### PR TITLE
docs: COPY examples to show how to use build-args

### PR DESCRIPTION
**Pull request by alexcb**
Signed-off-by: Alex Couture-Beil <alex@earthly.dev>

### CODE REVIEW

**Comment by alexcb** [comment]
This is related to #1605

File: docs/earthfile/earthfile.md [hunk 1]
```diff
@@ -442,6 +442,32 @@ COPY test/* .
...
+One can also copy from other Earthfile targets:
+
+```
+FROM alpine:3.13
+vault:
+    RUN echo aGVsbG8= > secret-data
+    SAVE ARTIFACT secret-data
+example:
+    COPY +vault/secret-data .
+    RUN cat secret-data | base64 -d
```
Comment (vladaionescu) [inline docs/earthfile/earthfile.md:454]: I would maybe not call this "secret-data", just so we don't encourage the placement of secrets in layers as part of the official docs. Perhaps use `dummy-artifact` instead?

Comment (alexcb) [reply docs/earthfile/earthfile.md:454]: lol.... yeah good point.

**Comment by alexcb** [comment]
example of examples working:
```

 1. Init 🚀
————————————————————————————————————————————————————————————————————————————————

           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)


 2. Build 🔧
————————————————————————————————————————————————————————————————————————————————

         alpine:3.13 | --> Load metadata linux/amd64
               +base | --> FROM alpine:3.13
               +base | [██████████] 100% resolve docker.io/library/alpine:3.13@sha256:026f721af4cf2843e07bba648e158fb35ecc876d822130633cc49f707f0fc88c
       +dummy-target | --> RUN echo aGVsbG8= > encoded-data
       +dummy-target | --> SAVE ARTIFACT encoded-data +dummy-target/encoded-data
            +example | --> COPY +dummy-target/encoded-data ./
            +example | --> RUN cat encoded-data | base64 -d
            +example | hello
              output | --> exporting outputs


 3. Push ⏫ (disabled)
————————————————————————————————————————————————————————————————————————————————

To enable pushing use

		earthly --push ...



 4. Local Output 🎁
————————————————————————————————————————————————————————————————————————————————



========================== 🌍 Earthly Build  ✅ SUCCESS ==========================

alex@mah:~/gh/earthly/earthly/docs$ earthly +example 

 1. Init 🚀
————————————————————————————————————————————————————————————————————————————————

           buildkitd | Found buildkit daemon as docker container (earthly-buildkitd)


 2. Build 🔧
————————————————————————————————————————————————————————————————————————————————

         alpine:3.13 | --> Load metadata linux/amd64
               +base | --> FROM alpine:3.13
               +base | [██████████] 100% resolve docker.io/library/alpine:3.13@sha256:026f721af4cf2843e07bba648e158fb35ecc876d822130633cc49f707f0fc88c
               +base | *cached* --> RUN apk add coreutils
       +dummy-target | encoder=base32
       +dummy-target | --> RUN echo hello | $encoder > encoded-data
       +dummy-target | encoder=base32
       +dummy-target | --> SAVE ARTIFACT encoded-data +dummy-target/encoded-data
            +example | --> COPY (encoder=base32) +dummy-target/encoded-data ./
            +example | --> RUN cat encoded-data | base32 -d
            +example | hello
              output | --> exporting outputs


 3. Push ⏫ (disabled)
————————————————————————————————————————————————————————————————————————————————

To enable pushing use

		earthly --push ...



 4. Local Output 🎁
```

**Review by vladaionescu — APPROVED** [review]

**Pull request merged**